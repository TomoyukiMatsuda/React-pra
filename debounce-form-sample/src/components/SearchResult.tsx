import React from "react";
import { IQiitaItem } from "../IQiitaItem";

interface Props {
  qiitaItems: IQiitaItem[] | null;
}

// 検索結果（フェッチしたアイテム）を表示するコンポーネント
export const SearchResult: React.VFC<Props> = (props) => {
  return (
    <div>
      {props.qiitaItems?.map((item) => {
        return (
          <div
            key={item.id}
            className="mb-3 py-2 px-8 bg-blue-100 rounded-lg shadow"
          >
            <p className="text-center font-bold  text-blue-800 mb-4 border-b-2 border-blue-800">
              {item.title}
            </p>
            <p className="text-blue-700">LGTM 👍：{item.likes_count}</p>
            <p className="text-blue-700">ユーザー：{item.user.name}</p>
          </div>
        );
      })}
    </div>
  );
};
