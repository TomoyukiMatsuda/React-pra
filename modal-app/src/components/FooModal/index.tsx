import React from "react";

interface Props {
  text: string;
}

export const FooModal: React.VFC<Props> = ({ text }) => {
  return (
    <div
      style={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        position: "fixed",
        top: "25%",
        left: "25%",
        zIndex: 20,
        height: "50%",
        width: "50%",
        boxShadow: "20px",
        background: "white",
        borderRadius: 30,
      }}
    >
      <p style={{ fontSize: 50 }}>{text}</p>
    </div>
  );
};
