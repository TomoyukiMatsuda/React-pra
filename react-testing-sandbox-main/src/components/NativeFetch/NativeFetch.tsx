import { usePokemons } from "~/hooks/usePokemons"

export function NativeFetch({ size }: { size: number }) {
  const [loading, pokemons, error] = usePokemons(size)
  if (error) {
    throw error
  }
  if (loading || pokemons === null) {
    return <p>loading...</p>
  }

  return (
    <>
      {pokemons.length === 0 ? (
        <p>no pokemon</p>
      ) : (
        <ul>
          {pokemons.map((pokemon) => {
            return (
              <li key={pokemon.id}>
                <img src={pokemon.image} width="120" />
                <p>{pokemon.name}</p>
              </li>
            )
          })}
        </ul>
      )}
    </>
  )
}
