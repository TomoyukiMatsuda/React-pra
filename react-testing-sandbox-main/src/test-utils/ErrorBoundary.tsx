import React from "react"

export class ErrorBoundary extends React.Component<unknown, { hasError: boolean; error: Error }> {
  state = { hasError: false, error: new Error() }
  static getDerivedStateFromError(error: Error) {
    return { hasError: true, error }
  }

  render() {
    if (this.state.hasError) {
      return <div data-testid="error">{this.state.error.message}</div>
    }
    return this.props.children
  }
}
