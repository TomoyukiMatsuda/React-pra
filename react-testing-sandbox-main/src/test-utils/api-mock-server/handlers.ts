import { graphql } from "msw"

export const handlers = [
  graphql.query("PokemonList", (req, res, ctx) => {
    const { size } = req.variables

    if (!size) {
      return res(
        ctx.data({
          pokemons: []
        })
      )
    }

    if (size === 400) {
      return res(
        ctx.status(400, ""),
        ctx.data({
          message: "This is 400."
        })
      )
    }

    return res(
      ctx.data({
        pokemons: [
          {
            id: "UG9rZW1vbjowMDE=",
            image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
            name: "Bulbasaur",
            __typename: "Pokemon"
          },
          {
            id: "UG9rZW1vbjowMDI=",
            image: "https://img.pokemondb.net/artwork/ivysaur.jpg",
            name: "Ivysaur",
            __typename: "Pokemon"
          }
        ]
      })
    )
  })
]
