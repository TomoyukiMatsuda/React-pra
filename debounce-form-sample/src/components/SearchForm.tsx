import React, { useState } from "react";
import Loader from "react-loader-spinner";
import { useDebounce } from "../hooks/useDebounce";
import { useFetchQiitaItems } from "../hooks/useFetchQiitaItems";
import { SearchResult } from "./SearchResult";

// 検索フォームコンポーネント（ここで debouncedValue を更新する）
export const SearchForm: React.VFC = () => {
  const [formText, setFormText] = useState<string>("");
  const debouncedValue = useDebounce(formText, 1000);
  const { qiitaItems, isLoading } = useFetchQiitaItems(debouncedValue);

  return (
    <>
      <label className="mt-12 mb-2 block text-gray-700 text-lg font-bold">
        Qiita 記事 検索キーワードを入力
      </label>
      <div className="flex w-full py-2 px-3 mb-4 shadow border rounded-full ">
        <input
          className="w-11/12 text-lg appearance-none border-none text-gray-700 focus:outline-none focus:shadow-outline"
          placeholder="例：React"
          value={formText}
          onChange={(e) => setFormText(e.target.value)}
        />
        <div className="h-6 w-6 ml-auto my-auto">
          <Loader
            type="Oval"
            color="#1E40AF"
            height={24}
            width={24}
            visible={isLoading}
          />
        </div>
      </div>
      <SearchResult qiitaItems={qiitaItems} />
    </>
  );
};
