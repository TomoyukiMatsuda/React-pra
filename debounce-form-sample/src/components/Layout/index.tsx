import React, { useState } from "react";
import { useDebounce } from "../../hooks/useDebounce";
import { useFetchQiitaItems } from "../../hooks/useFetchQiitaItems";
import { SearchResult } from "./SearchResult";
import { SearchForm } from "./SearchForm";

export const Layout: React.VFC = () => {
  const [formText, setFormText] = useState<string>("");
  const debouncedValue = useDebounce(formText, 1000);
  const { fetchQiitaItems, qiitaItems, isLoading } =
    useFetchQiitaItems(debouncedValue);

  return (
    <>
      <div className="mt-12 mb-8">
        <SearchForm
          formText={formText}
          setFormText={setFormText}
          isLoading={isLoading}
        />
      </div>
      <SearchResult fetchQiitaItems={fetchQiitaItems} qiitaItems={qiitaItems} />
    </>
  );
};
