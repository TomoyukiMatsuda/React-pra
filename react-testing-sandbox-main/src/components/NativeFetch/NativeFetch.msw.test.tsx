import { cleanup, render, screen, waitFor, waitForElementToBeRemoved } from "@testing-library/react"

import { NativeFetch } from "~/components/NativeFetch"
import { server } from "~/test-utils/api-mock-server/server"
import { ErrorBoundary } from "~/test-utils/ErrorBoundary"

const WrappedNativeFetch = ({ size }: { size: number }) => {
  return (
    <ErrorBoundary>
      <NativeFetch size={size} />
    </ErrorBoundary>
  )
}

describe("NativeFetch", () => {
  beforeAll(() => {
    server.listen()
  })
  afterEach(() => {
    cleanup()
    server.resetHandlers()
  })
  afterAll(() => {
    server.close()
  })
  test("render:loding", async () => {
    const { asFragment } = render(<WrappedNativeFetch size={5} />)
    screen.getByText("loading...")
    expect(asFragment()).toMatchSnapshot()
    await waitForElementToBeRemoved(() => screen.getByText("loading..."))
  })
  test("render:pokemons", async () => {
    const { asFragment } = render(<WrappedNativeFetch size={5} />)
    screen.getByText("loading...")
    await waitForElementToBeRemoved(() => screen.getByText("loading..."))
    expect(asFragment()).toMatchSnapshot()
  })
  test("render:no pokemon", async () => {
    const { asFragment } = render(<WrappedNativeFetch size={0} />)
    await waitFor(() => screen.getByText("no pokemon"))
    expect(asFragment()).toMatchSnapshot()
  })
  test("error", async () => {
    const spy = jest.spyOn(console, "error")
    spy.mockImplementation(() => void 0)
    render(<WrappedNativeFetch size={400} />)
    await waitFor(() => screen.getByText("http status: 400"))
    spy.mockRestore()
  })
})
