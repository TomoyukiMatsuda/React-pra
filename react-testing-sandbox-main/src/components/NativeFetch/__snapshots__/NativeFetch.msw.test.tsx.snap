// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeFetch render:loding 1`] = `
<DocumentFragment>
  <p>
    loading...
  </p>
</DocumentFragment>
`;

exports[`NativeFetch render:no pokemon 1`] = `
<DocumentFragment>
  <p>
    no pokemon
  </p>
</DocumentFragment>
`;

exports[`NativeFetch render:pokemons 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <img
        src="https://img.pokemondb.net/artwork/bulbasaur.jpg"
        width="120"
      />
      <p>
        Bulbasaur
      </p>
    </li>
    <li>
      <img
        src="https://img.pokemondb.net/artwork/ivysaur.jpg"
        width="120"
      />
      <p>
        Ivysaur
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
